core/src/main/java/org/mskcc/cbio/oncokb/util/CacheUtils.java

```diff
@@ -253,10 +254,87 @@ private static void cacheAllGenes() {
             genesByEntrezId.put(gene.getEntrezGeneId(), gene);
             hugoSymbolToEntrez.put(gene.getHugoSymbol(), gene.getEntrezGeneId());
         }
-
+        cancerGeneList = null;
         System.out.println("Cache all genes: " + MainUtils.getTimestampDiff(current) + " at " + MainUtils.getCurrentTime());
     }
 
+    public static List<CancerGene> getCancerGeneList() {
+        if(cancerGeneList == null) {
+            updateCancerGeneList();
+        }
+        return cancerGeneList;
+    }
+
+    private static void updateCancerGeneList() {
+        cancerGeneList = new ArrayList<>();
+        try {
+            List<String> lines = FileUtils.readTrimedLinesStream(
+                CancerGeneUtils.class.getResourceAsStream(CANCER_GENE_FILE_PATH));
+            Iterator itr = lines.iterator();
+            Set<String> allHugoSymbolsFromFile = new HashSet<>();
+
+            while (itr.hasNext()) {
+                String line = itr.next().toString().trim();
+                // skip comments
+                if (line.startsWith("#")) {
+                    continue;
+                }
+
+                String[] items = line.split("\t");
+                if (items.length != 9) continue;
+
+                String hugoSymbol = items[0];
+                allHugoSymbolsFromFile.add(hugoSymbol);
+
+                Gene gene = GeneUtils.getGeneByEntrezId(Integer.parseInt(items[1]));
+                CancerGene cancerGene = new CancerGene();
+                cancerGene.setEntrezGeneId(Integer.parseInt(items[1]));
+
+                if (gene == null) {
+                    cancerGene.setHugoSymbol(hugoSymbol);
+                } else {
+                    if (!gene.getHugoSymbol().equals(hugoSymbol)) {
+                        System.out.println("The gene hugo does not match, expect " + gene.getHugoSymbol() + ", but got: " + hugoSymbol);
+                    }
+                    cancerGene.setHugoSymbol(gene.getHugoSymbol());
+                    cancerGene.setOncokbAnnotated(true);
+                    cancerGene.setOncogene(gene.getOncogene());
+                    cancerGene.setTSG(gene.getTSG());
+                }
+                int occurence = NumberUtils.isNumber(items[2].trim()) ? Integer.parseInt(items[2].trim()) : 0;
+                if (cancerGene.getOncokbAnnotated()) {
+                    occurence++;
+                }
+                cancerGene.setOccurrenceCount(occurence);
+                cancerGene.setmSKImpact(items[3].trim().equals("1"));
```

Comment:
What does `equals("1")` mean in here?  Do you mean the length of element after trim()?

Reply:
This is the actual value from the file. In the file, this column, 1 means exists